src/schematools/datasetcollection.py

```diff
@@ -18,27 +24,35 @@ class DatasetCollection(metaclass=Singleton):
     """
 
     def __init__(self) -> None:
-        self.datasets_cache = {}
-
-    def _load_dataset(self, dataset_id: str) -> Optional[DatasetSchema]:
-        """Loads the dataset from SCHEMA_URL.
+        """Initialize the DatasetCollection.
 
-        If SCHEMA_URL is not defined, return None.
+        Args:
+            schema_loader: An alternative schema loader can be provided
+                If schema_loader is None, the default url loader is used.
         """
-        # Avoid circular import problem.
-        from schematools.utils import dataset_schema_from_url
+        self.datasets_cache: Dict[str, DatasetSchema] = {}
+        self.schema_loader: Optional[loaders.SchemaLoader] = None
```

Comment:
Maybe these should be defined as instance variables according to https://www.python.org/dev/peps/pep-0526/

Reply:
Shouldn't these also be private, `_datasets_cache` etc.?